.github/workflows/publish-package.yml

```
@@ -26,10 +26,11 @@ jobs:
       run: |
         python -m pep517.build --source --binary --out-dir dist/ .
     - name: Publish distribution 📦 to Test PyPI
-      if: github.event_name == 'push'
+      if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```

As I think this is the most confusing part of the logic, above this can we have the comment?
```suggestion
      # every PR will trigger a push event on master, so check the push event is actually coming from master
      if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```